pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -195,8 +196,10 @@ func (self *RefsHelper) NewBranch(from string, fromFormattedName string, suggest
 	})
 }
 
-// sanitizedBranchName will remove all spaces in favor of a dash "-" to meet
+// sanitizedBranchName will remove all spaces in favor of a replacement char to meet
 // git's branch naming requirement.
-func sanitizedBranchName(input string) string {
-	return strings.Replace(input, " ", "-", -1)
+func (self *RefsHelper) sanitizedBranchName(input string) string {
+	reg1 := regexp.MustCompile(`^-+`)
+	reg2 := regexp.MustCompile(`\.|\/\.|\.\.|~|\^|:|\/$|\.lock$|\.lock\/|\\|\*|\s|^\s*$|\.$|\[|\]$`)
```

Comment:
This regex feels like overkill to me. I would just have a slice of characters that we want to replace with hyphens one-to-one. The `.lock` thing is too presumptuous so let's remove that.

Reply:
I copied this regex from vscode, but I can remove that part.